src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -509,7 +511,7 @@ void GcsPlacementGroupManager::CleanPlacementGroupIfNeededWhenActorDead(
       continue;
     }
     placement_group->MarkCreatorActorDead();
-    if (placement_group->IsPlacementGroupRemovable()) {
+    if (!placement_group->IsDetached() && placement_group->IsPlacementGroupRemovable()) {
```

Comment:
@ffbin @rkooo567  Do you think we should keep the pg alive when job dead if the pg is detached ?  I think we don't have the requirement temporarily ?

Reply:
No. Detached means the placement group is detached from the job, so it shouldn't be removed!

Reply:
yeah. I will update this.